packages/venia-concept/src/components/ProductImageCarousel/thumbnailList.js

```diff
@@ -11,18 +12,39 @@ class ThumbnailList extends Component {
         classes: PropTypes.shape({
             root: PropTypes.string
         }),
-        items: PropTypes.arrayOf(
-            PropTypes.shape({
-                label: PropTypes.string,
-                position: PropTypes.number,
-                disabled: PropTypes.bool,
-                file: PropTypes.string.isRequired
-            })
-        ).isRequired
+        items: PropTypes.arrayOf(imageItemPropType).isRequired,
+        getItemKey: PropTypes.func,
+        activeItemSrc: PropTypes.string,
+        updateActiveItemIndex: PropTypes.func
+    };
+
+    findItemIndexBySrc = src =>
+        this.props.items.findIndex(image => image.file === src);
+
+    updateActiveItemHandler = newActiveItemSrc => {
+        const newActiveItemIndex = this.findItemIndexBySrc(newActiveItemSrc);
+        this.props.updateActiveItemIndex(newActiveItemIndex);
     };
 
     render() {
-        return <List renderItem={Thumbnail} {...this.props} />;
+        const { items, getItemKey, activeItemSrc, classes } = this.props;
+
+        return (
+            <List
+                items={items}
+                renderItem={props => (
+                    <Thumbnail
+                        {...props}
+                        activeItemSrc={activeItemSrc}
+                        onClickHandler={() =>
```

Comment:
Instead of creating separate handler for each Thumbnail after every rendering it is better just to allow Thumbnail element to pass this information when it is clicked.

Reply:
good concern, thanks